### ISSUE TITLE
Style/EmptyLiteral false positive.

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#13189]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

When you have `Hash.new([])` (hash constructor with default value), it should not trigger `Style/EmptyLiteral`

## Actual behavior

Describe here what actually happened.

It corrects it to `{}`.  

This is not equivalent code. For example:

```
irb(main):001> a = Hash.new([])
=> {}
irb(main):002> a[:b]
=> []
irb(main):003> x = {}
=> {}
irb(main):004> x[:b]
=> nil
```

## Steps to reproduce the problem
Put `Hash.new([])` in your code

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V

1.66.0 (using Parser 3.3.4.2, rubocop-ast 1.32.2, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-capybara 2.21.0
  - rubocop-factory_bot 2.26.1
  - rubocop-rails 2.26.0
  - rubocop-rspec 3.0.4
  - rubocop-rspec_rails 2.30.0
```

**Comment by koic** [comment]
This issue is a dup with https://github.com/rubocop/rubocop/issues/13181.
